vpr/src/place/place.cpp

```diff
@@ -1463,21 +1635,64 @@ static e_move_result try_swap(float t,
         //Update the block positions
         apply_move_blocks(blocks_affected);
 
-        // Find all the nets affected by this swap and update their costs
+        //Find all the nets affected by this swap and update their costs
+        //This routine calculates new connection delays and timing costs
+        //and store them in proposed_* data structures
+        //This routine also calculates the wiring cost, which doesn't
+        //depend on the timing driven data
         int num_nets_affected = find_affected_nets_and_update_costs(place_algorithm,
                                                                     delay_model,
                                                                     criticalities,
                                                                     blocks_affected,
                                                                     bb_delta_c,
                                                                     timing_delta_c);
-        if (place_algorithm == PATH_TIMING_DRIVEN_PLACE) {
+
+        //Find all the sink pins with changed connection delays from the affected blocks.
+        //These sink pins will be passed into the pin_timing_invalidator for timing update.
+        //They will also be added to the pin invalidator when we wish to revert a timing update.
+        std::vector<ClusterPinId> sink_pins_affected;
+        find_affected_sink_pins(blocks_affected, sink_pins_affected);
+
+        if (place_algorithm == SLACK_TIMING_PLACE) {
```

Comment:
I think an overview comment before this big if would be helpful.  The blocks have been temporarily moved, and now we're going to assess cost. We need to update all timing etc. information that would impact the cost, but remember what we did and undo if we reject the move.